core/src/main/scala/com/microsoft/azure/synapse/ml/explainers/ICEExplainer.scala

```diff
@@ -0,0 +1,252 @@
+      .map(row => row.get(0))
+    values
+  }
+
+  private def createNSplits(n: Int)(from: Double, to: Double): Seq[Double] = {
+    (0 to n) map {
+      i => (to - from) / n * i + from
+    }
+  }
+
+  private def collectSplits(df: DataFrame, numericFeature: ICENumericFeature): Array[Double] = {
+    val (feature, nSplits, rangeMin, rangeMax) = (numericFeature.name, numericFeature.getNumSplits,
+      numericFeature.rangeMin, numericFeature.rangeMax)
+    val featureCol = df.schema(feature)
+
+    val createSplits = createNSplits(nSplits) _
+
+    val values = if (rangeMin.isDefined && rangeMax.isDefined) {
+      val (mi, ma) = (rangeMin.get, rangeMax.get)
+      // The ranges are defined
+      featureCol.dataType match {
+        case _@(ByteType | IntegerType | LongType | ShortType) =>
+          if (ma.toLong - mi.toLong <= nSplits) {
+            // For integral types, no need to create more splits than needed.
+            (mi.toLong to ma.toLong) map (_.toDouble)
+          } else {
+            createSplits(mi, ma)
+          }
+        case _ =>
+          createSplits(mi, ma)
+      }
+    } else {
+      // The ranges need to be calculated from background dataset.
+      featureCol.dataType match {
+        case _@(ByteType | IntegerType | LongType | ShortType) =>
+          val Row(minValue: Long, maxValue: Long) = df
+            .agg(min(col(feature)).cast(LongType), max(col(feature)).cast(LongType))
+            .head
+
+          val mi = rangeMin.map(_.toLong).getOrElse(minValue)
+          val ma = rangeMax.map(_.toLong).getOrElse(maxValue)
+
+          if (ma - mi <= nSplits) {
+            // For integral types, no need to create more splits than needed.
+            (mi to ma) map (_.toDouble)
+          } else {
+            createSplits(mi, ma)
+          }
+        case _ =>
+          val Row(minValue: Double, maxValue: Double) = df
+            .agg(min(col(feature)).cast(DoubleType), max(col(feature)).cast(DoubleType))
+            .head
+
+          val mi = rangeMin.getOrElse(minValue)
+          val ma = rangeMax.getOrElse(maxValue)
+          createSplits(mi, ma)
+      }
+    }
+    values.toArray
+  }
+
+  override def copy(extra: ParamMap): Transformer = this.defaultCopy(extra)
+
+  override def transformSchema(schema: StructType): StructType = {
+    // Check the data type for categorical features
+    val categoricalFeaturesTypes= getCategoricalFeatures.map(f => schema(f.name).dataType)
+    val allowedCategoricalTypes = Array(StringType, BooleanType, ByteType, ShortType, IntegerType, LongType)
+    require(categoricalFeaturesTypes.forall(allowedCategoricalTypes.contains),
+      s"Data type for categorical features must be ${allowedCategoricalTypes.mkString("[", ",", "]")}.")
+
+    // Check the data type for numeric features
+    val numericFeaturesTypes= getNumericFeatures.map(f => schema(f.name).dataType)
+    val allowedNumericTypes = Array(FloatType, DoubleType, DecimalType)
```

Comment:
Add ByteType, ShortType, IntegerType, LongType

Reply:
[A similar check happens in data balance analysis validateSchema, check it out here](https://github.com/microsoft/SynapseML/blob/ff2aa020693b6521da988250376557bf128301ef/core/src/main/scala/com/microsoft/azure/synapse/ml/exploratory/DataBalanceParams.scala#L37)